### ISSUE TITLE
[Blazor] AddInteractiveWebAssemblyComponents() should register LazyAssemblyLoader or a similar type.

### ISSUE THREAD

**Issue by javiercn** [dotnet/aspnetcore#51966]
* Otherwise, lazy assembly loading breaks during prerendering.
* The workaround/fix is to register the service yourself.
  * builder.Services.AddScoped<LazyAssemblyLoader>();

This never worked in the past with webassembly, but it was not as prevalent as we were not doing prerendering.

We can start by registering a mock implementation inside AddInteractiveWebAssemblyComponents, adding a reference is not suitable in a patch release.

**Comment by javiercn** [comment]
We should consider fixing this issue and https://github.com/dotnet/aspnetcore/issues/26601 at the same time.

**Comment by javiercn** [comment]
I don't think we can patch this. It will require us to add a reference to `Components.WebAssembly` which is changing public API, so I think the best we can do is tell people to register the type themselves.

**Comment by mkArtakMSFT** [comment]
> I don't think we can patch this. It will require us to add a reference to `Components.WebAssembly` which is changing public API, so I think the best we can do is tell people to register the type themselves.

Sounds like we need to document this as a known issue then?

**Comment by oroztocil** [comment]
Fixed by https://github.com/dotnet/aspnetcore/pull/61822
